UaClient/ServiceModel/Ua/EncodingDictionary.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace Workstation.ServiceModel.Ua
+{
+    public class EncodingDictionary
+    {
+        private readonly Dictionary<NodeId, Type> encodingIdToTypeDictionary = new Dictionary<NodeId, Type>();
+        private readonly Dictionary<Type, NodeId> typeToEncodingIdDictionary = new Dictionary<Type, NodeId>();
+
+        public EncodingDictionary(IEnumerable<(ExpandedNodeId,Type)> table, IList<string> namespaceUris)
+        {
+            foreach (var (nodeId, type) in table)
+            {
+                ushort ns = nodeId.NodeId.NamespaceIndex;
+                var nsu = nodeId.NamespaceUri;
+                if (!string.IsNullOrEmpty(nsu))
+                {
+                    var i = namespaceUris.IndexOf(nsu!);
+                    if (i == -1)
+                    {
+                        continue;
+                    }
+                    ns = (ushort)i;
+                }
+
+                var encodingId = nodeId.NodeId.IdType switch
+                {
+                    IdType.Numeric  => new NodeId((uint)nodeId.NodeId.Identifier, ns),
+                    IdType.String   => new NodeId((string)nodeId.NodeId.Identifier, ns),
+                    IdType.Guid     => new NodeId((Guid)nodeId.NodeId.Identifier, ns),
+                    _               => new NodeId((byte[])nodeId.NodeId.Identifier, ns)
+                };
+
+                this.encodingIdToTypeDictionary[encodingId] = type;
+                this.typeToEncodingIdDictionary[type] = encodingId;
+            }
+        }
+        
+        public EncodingDictionary(IEnumerable<(ExpandedNodeId,Type)> table)
+        {
+            foreach (var (nodeId, type) in table)
+            {
+                ushort ns = nodeId.NodeId.NamespaceIndex;
+                var nsu = nodeId.NamespaceUri;
+                if (!string.IsNullOrEmpty(nsu))
+                {
+                    continue;
```

Comment:
This constructor is called before we retrieved the URI list. Should we perform more checks to be sure that this node id/type pair does (not) belongs to the OPC UA namespace?